src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -108,7 +108,13 @@ public RoomDto playAgain(String roomId, PlayAgainRequest request) {
             throw new ApiException(GameError.GAME_NOT_OVER);
         }
 
-        Room room = game.getRoom();
+        // Get up to date room using repository
+        Room room = repository.findRoomByRoomId(game.getRoom().getRoomId());
```

Comment:
When users leave the room via the removeUser endpoint, it updates the room object in the database. However, the game object still holds the original room reference, so when playing again, it should fetch from the database instead (otherwise on the lobby page all the users who left would show up again)

Reply:
Sounds good, thanks for the explanation! Is there any reason we may need to keep the room reference on the game object updated - shouldn't this reference be up-to-date with the latest database information, or no?

Reply:
As of now the game object's room is updated every time a player's active status changes - keeping it updated when a player leaves the room means we would probably need a conditional check in the leaveRoom endpoint to check if the game exists and then manually update the game (since the game object's room is just a variable and not synced with the actual database). Which tbf I think would be good to add - I think maybe when we implement the notification on player exit from the game would be a good time to implement this if that's cool